pkg/spec/spec.go

```diff
@@ -48,6 +50,30 @@ func supercedeUserMounts(mounts []spec.Mount, configMount []spec.Mount) []spec.M
 	return configMount
 }
 
+// Split named volumes from normal volumes
+func splitNamedVolumes(mounts []spec.Mount) ([]spec.Mount, []*libpod.ContainerNamedVolume) {
+	newMounts := make([]spec.Mount, 0)
+	namedVolumes := make([]*libpod.ContainerNamedVolume, 0)
+	for _, mount := range mounts {
+		// If it's not a named volume, append unconditionally
+		if mount.Type != bindMount {
+			newMounts = append(newMounts, mount)
+			continue
+		}
+		// Volumes that are not named volumes must be an absolute or relative path
+		if strings.Contains(mount.Source, "/") || strings.Contains(mount.Source, ".") {
```

Comment:
it seems that Docker doesn't accept relative paths, but accept named volumes as `foo.bar`.  If we want to extend it in a non breaking way, we could limit relative paths to having a prefix like `./`, but raise an error on, e.g. `-v .foo:/foo`

What do you think?

Reply:
Right now we're using the same name regex for volumes that we do for pods and containers - letters, numbers, underscores only. I need to check and see what Docker uses for their volume name regex.

I don't know if I mind requiring that volume names not allow `.` if it makes detecting/allowing relative paths easier. @rhatdan @giuseppe WDYT?

Reply:
I don't like the idea of not allowing volume names that Docker does allow.  

We can document that we allow paths that begin with . and / only.

Reply:
Alright, looking at this, it seems like Docker allows `.` in volume and container names now. No other changes I could see.

Will modify our regex in a separate PR to handle this, then update this PR.

Reply:
Fixed. We now use HasPrefix() to check only the first character which is guaranteed to be alphanumeric for names.